edumanage/models.py

```diff
@@ -21,109 +42,157 @@ class MultiSelectFormField(forms.MultipleChoiceField):
     widget = forms.CheckboxSelectMultiple
 
     def __init__(self, *args, **kwargs):
-        self.max_choices = 4
-        super(MultiSelectFormField, self).__init__(*args, **kwargs)
-
-    def clean(self, value):
-        if not value and self.required:
-            raise forms.ValidationError(self.error_messages['required'])
-        # if value and self.max_choices and len(value) > self.max_choices:
-        #     raise forms.ValidationError('You must select a maximum of %s choice%s.'
-        #             % (apnumber(self.max_choices), pluralize(self.max_choices)))
-        return value
-
+        supercls = super(MultiSelectFormField, self)
+        # remove TypedChoiceField extra args
+        supercls_init_args = getargspec(supercls.__init__).args[1:]
+        for key in list(kwargs):
+            if key not in supercls_init_args:
+                del kwargs[key]
+        supercls.__init__(*args, **kwargs)
 
 class MultiSelectField(models.Field):
+    _separator_default = ','
 
-    def get_internal_type(self):
-        return "CharField"
+    def __init__(self, *args, **kwargs):
+        self.separator = kwargs.pop(
+            'separator', self._separator_default)
+        super(MultiSelectField, self).__init__(*args, **kwargs)
 
-    def get_choices_default(self):
-        return self.get_choices(include_blank=False)
+    def deconstruct(self):
+        name, path, args, kwargs = super(
+            MultiSelectField, self).deconstruct()
+        if self.separator != self._separator_default:
```

Comment:
Django docs [suggest the deconstructor should only return kwargs that don't carry the default value](https://docs.djangoproject.com/en/1.11/howto/custom-model-fields/#field-deconstruction) (as they are deemed redundant otherwise).